lib/src/main/java/grpcbridge/parser/Deserializer.java

```diff
@@ -1,11 +1,12 @@
 package grpcbridge.parser;
 
+import com.google.common.net.MediaType;
 import com.google.protobuf.Message;
 import grpcbridge.http.HttpRequest;
 import grpcbridge.rpc.RpcMessage;
 
 public interface Deserializer {
     RpcMessage deserialize(HttpRequest httpRequest, Message.Builder builder);
 
-    boolean supported(String contentType);
+    boolean supported(MediaType contentType);
```

Comment:
Yeah so the supported doesn't do a character encoding check, it only checks the type and subtype values, the reason why i'm using this instead is because i had issues before where if the content-type came through with additional information like charset, the string comparison would fail. This just makes it simpler and more predictable. And yes the charset is passed to the deserialize method